main/realmain.cc

```diff
@@ -718,22 +723,23 @@ int realmain(int argc, char *argv[]) {
                     // Calculate file hashes alongside indexing when --store-state is specified for LSP mode
                     ? hashing::Hashing::indexAndComputeFileHashes(*gs, opts, *logger, inputFilesSpan, *workers, kvstore)
                     : pipeline::index(*gs, inputFilesSpan, opts, *workers, kvstore);
+            ENFORCE(nonPackageIndexed.hasResult(), "There's no cancellation in batch mode");
```

Comment:
Should we call `nonPackagedIndexed` something like `nonPackagedIndexedResult` and then assign

```cpp
auto nonPackagedIndexed = move(nonPackagedIndexed.result());
```
after the ENFORCE sot hat we don't have to sprinkle the `.result()` calls throughout the rest of the method?